test/index.test.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-env mocha */
+
+'use strict'
+
+const assert = require('assert')
+const { promisify } = require('util')
+
+const createGrapes = require('bfx-svc-test-helper/grapes')
+const createFxGrenache = require('bfx-svc-test-helper/fauxgrenache')
+
+const Fac = require('..')
+
+const stubs = {
+  'rest:util:net': {
+    getIpInfo (space, ip, cb) {
+      const res = [ip, { country: 'US', region: 'CA' }]
+      return cb(null, res)
+    }
+  }
+}
+
+const grapes = createGrapes()
+const caller = {
+  on: () => {},
+  ctx: { root: __dirname }
+}
+
+const fac = new Fac(caller, {})
+fac.tick = () => {}
+fac.conf = { grape: 'http://127.0.0.1:30001', transport: 'http' }
+
+let fxg
+describe('RPC integration', () => {
+  before(async function () {
+    this.timeout(20000)
+
+    grapes.onAnnounce()
+
+    await grapes.start()
+    fxg = createFxGrenache(stubs, grapes)
+    await fxg.start()
+
+    const start = promisify(fac.start).bind(fac)
+    await start()
+  })
+
+  after(async function () {
+    this.timeout(5000)
+
+    await grapes.stop()
+    await fxg.stop()
+    fac.stop()
+  })
+
+  describe('_doReq', () => {
+    describe('callback', () => {
+      it('options defaults', done => {
+        fac._doReq('request', 'rest:util:net', 'getIpInfo', ['8.8.8.8'], (_err, res) => {
```

Comment:
why not testing `.req` and `.map`, the methods that are used?

Reply:
because both `req` and `map` call `_doReq`. I can add small tests for `map` and `req` to ensure they're calling `_doReq` with proper params